Framework/Algorithms/src/RealFFT.cpp

```diff
@@ -173,13 +181,14 @@ void RealFFT::exec() {
     gsl_fft_halfcomplex_wavetable_free(wavetable);
     gsl_fft_real_workspace_free(workspace);
 
+    auto &y = outWS->mutableY(0);
     for (int i = 0; i < yOutSize; i++) {
       double x = df * i;
-      outWS->dataX(0)[i] = x;
-      outWS->dataY(0)[i] = data[i] / df;
+      xData[i] = x; // xData used from above
+      y[i] = data[i] / df;
     }
     if (outWS->isHistogramData())
-      outWS->dataX(0)[yOutSize] = outWS->dataX(0)[yOutSize - 1] + df;
+      outWS->mutableX(0)[yOutSize] = outWS->mutableX(0)[yOutSize - 1] + df;
     // outWS->getAxis(1)->spectraNo(0)=inWS->getAxis(1)->spectraNo(spec);
```

Comment:
@DimitarTasev Why does this code need `data` to be a `boost::shared_array<double>`? Can't you just directly let the code above modify the data via `outWS->mutableY` and then just divide by `df` in the end?